### PR TITLE
Fix dependency issues and update Dockerfile

### PR DESCRIPTION
**Pull request by alexgleith**
* Fix issue with Jinja imports changing
* Redo the Dockerfile to update to the new geobase pattern
* Fix some dependencies, remove others, simplify.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -57,8 +57,8 @@
         "cachetools",
         "click",
         "dataclasses>=0.6;python_version<'3.7'",
-        "datacube>=1.8",
-        "eodatasets3 >= 0.17.0",
+        "datacube",
+        "eodatasets3",
```
Comment (jeremyh) [inline setup.py]: These are still valid, aren't they? They're essential minimums

Comment (jeremyh) [reply setup.py]: They'll still be relevant in the setup.py for regular pip installations (... and for when it goes to pypi)

Comment (alexgleith) [reply setup.py]: Hmm, I guess so. I get confused as to when to include in setup.py vs requirements.txt vs constraints.txt still! I'll put them back.

**Review by jeremyh — APPROVED** [review]
Thanks for fixing these! 🙂

**Pull request merged**